### PR TITLE
[JENKINS-31458] Taking advantage of nested object construction in StepConfigTester

### PR DESCRIPTION
**Pull request by jglick**
Based on #229 (just to get everything set up for Jenkins dev builds) but downstream of https://github.com/jenkinsci/jenkins/pull/1936.

[Incremental diff](https://github.com/jglick/workflow-plugin/compare/JobPropertyStep+OptionalJobProperty+BuildDiscarderProperty...DataBoundConstructor-JENKINS-31458)

@reviewbybees

### CODE REVIEW

File: support/src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java [hunk 1]
```diff
@@ -87,7 +88,7 @@
                 ConsoleLogFilter filter = get(ConsoleLogFilter.class);
                 OutputStream os = new FileOutputStream(la.getLogFile(), true);
                 if (filter != null) {
-                    os = filter.decorateLogger(null, os);
+                    os = filter.decorateLogger((AbstractBuild) null, os);
```
Comment (amuniz) [inline support/src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java]: `Run`?

File: support/src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java [hunk 2]
```diff
@@ -27,6 +27,7 @@
 import org.jenkinsci.plugins.workflow.steps.StepConfigTester;
 import org.junit.Test;
 import static org.junit.Assert.*;
+import org.junit.Ignore;
```
Comment (amuniz) [inline support/src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java]: Not used?

File: support/src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java (hunk 1 above)
Comment (jglick) [inline support/src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java]: No.

This part of the patch is just part of the `post-625` branch, not this PR specifically.

File: support/src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java (hunk 2 above)
Comment (jglick) [inline support/src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java]: Also just part of the `post-625` branch.

**Comment by amuniz** [comment]
:bee:

**Comment by jglick** [comment]
Test failures might be from picking up the wrong core snapshot, not sure.

**Comment by jglick** [comment]
Folding into #321.

**Pull request closed without merging**